python/dolfin/fem/assembling.py

```diff
@@ -39,10 +38,10 @@ def _create_cpp_form(form, form_compiler_parameters=None,
                 "Ignoring form_compiler_parameters when passed a dolfin Form!")
         return form
     elif isinstance(form, ufl.Form):
-        return Form(
+        form = Form(
```

Comment:
Change this to `form = fem.Form` (and modify the above imports to reflect the change).

This will help emphasise distinction between `ufl.Form`, `dolfin.fem.(form).Form` and `dolfin.cpp.fem.Form`.